src/app/add-transaction.component.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { AddExpenseComponent } from './add-transaction.component';
+
+describe('AddExpenseComponent', () => {
+    let fixture: ComponentFixture<AddExpenseComponent>;
+    let component: AddExpenseComponent;
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [AddExpenseComponent],
+    }).compileComponents();
+    fixture = TestBed.createComponent(AddExpenseComponent);
+    component = fixture.componentInstance;
+  });
+  it('should hide modal in cancel', () => {
+     component.cancel()
+     expect(component.open()).toBe(false);
+     expect(component.type()).toBe('expense');
+  });
+  it('should test addTransaction', () => {
```

Comment:
Could have been more explicit here, like "should not submit when invalid and submit the valid amount"